src/collaborators.rs

```diff
@@ -0,0 +1,91 @@
+extern crate serde_json;
+
+use rep::User;
+use self::super::{Github, Result, Error};
+use hyper::status::StatusCode;
+
+use std::collections::HashMap;
+use std::fmt;
+
+#[derive(Debug)]
+pub struct Permissions {
+    pub admin: bool,
+    pub push: bool,
+    pub pull: bool,
+}
+
+impl Default for Permissions {
+    fn default() -> Permissions {
+        Permissions {
+            admin: false,
+            push: true,
+            pull: false,
+        }
+    }
+}
+
+impl fmt::Display for Permissions {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f,
+               "{}",
+               if self.admin {
+                   "admin"
+               } else if self.pull {
+                   "pull"
+               } else { // push is the default
+                   "push"
+               }
+        )
+    }
+}
+
+pub struct Collaborators<'a> {
+    github: &'a Github<'a>,
+    owner: String,
+    repo: String,
+}
+
+impl<'a> Collaborators<'a> {
+    pub fn new<O, R>(github: &'a Github<'a>, owner: O, repo: R) -> Collaborators<'a>
+        where O: Into<String>,
+              R: Into<String>
+    {
+        Collaborators {
+            github: github,
+            owner: owner.into(),
+            repo: repo.into(),
+        }
+    }
+
+    fn path(&self, more: &str) -> String {
+        format!("/repos/{}/{}/collaborators{}", self.owner, self.repo, more)
+    }
+
+    pub fn list(&self) -> Result<Vec<User>> {
+        self.github.get::<Vec<User>>(&self.path(""))
+    }
+
+    pub fn is_collaborator(&self, username: &str) -> Result<bool> {
+        match self.github.get::<()>(&self.path(&format!("/{}", username))) {
+            Ok(_) => Ok(true),
+            Err(Error::Fault { code: c, .. }) if c == StatusCode::NotFound => Ok(false),
+            Err(other) => Err(other),
+        }
+    }
+
+    pub fn add(&self, username: &str, permissions: &Permissions) -> Result<()> {
```

Comment:
It's a bit weird that this gets passed in as an object representing set of permissions when really you are setting a [single permission](https://developer.github.com/v3/repos/collaborators/#add-user-as-a-collaborator), `admin`, `push`, or `pull`. What do you think about the idea of a Permission enum with Admin, Push, and Pull variants. Something like [issues::Filter](https://github.com/softprops/hubcaps/blob/master/src/issues.rs#L10-L37) may be a better model for this